Tests error with cornice 3.4.x: ConfigurationConflictError: Conflicting configuration actions

Fresh clone, master branch, running tox:

```
GLOB sdist-make: /home/churchyard/tmp/cornice.ext.sphinx/setup.py
py27 create: /home/churchyard/tmp/cornice.ext.sphinx/.tox/py27
py27 installdeps: nose, mock, Sphinx
py27 inst: /home/churchyard/tmp/cornice.ext.sphinx/.tox/dist/cornice_sphinx-0.4.dev0.zip
py27 installed: alabaster==0.7.12,Babel==2.6.0,certifi==2018.11.29,chardet==3.0.4,cornice==3.4.3,cornice-sphinx==0.4.dev0,docutils==0.14,funcsigs==1.0.2,hupper==1.4.2,idna==2.7,imagesize==1.1.0,Jinja2==2.10,MarkupSafe==1.1.0,mock==2.0.0,nose==1.3.7,packaging==18.0,PasteDeploy==2.0.0,pbr==5.1.1,plaster==1.0,plaster-pastedeploy==0.6,Pygments==2.3.0,pyparsing==2.3.0,pyramid==1.10.1,pytz==2018.7,repoze.lru==0.7,requests==2.20.1,simplejson==3.16.0,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.8.2,sphinxcontrib-websupport==1.1.0,translationstring==1.3,typing==3.6.6,urllib3==1.24.1,venusian==1.1.0,WebOb==1.8.4,zope.deprecation==4.4.0,zope.interface==4.6.0
py27 run-test-pre: PYTHONHASHSEED='3467803108'
py27 runtests: commands[0] | python --version
Python 2.7.15
py27 runtests: commands[1] | nosetests
..E.
======================================================================
ERROR: test_string_validator_resolved (tests.TestServiceDirective)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/churchyard/tmp/cornice.ext.sphinx/tests/__init__.py", line 48, in test_string_validator_resolved
    ret = self.directive.run()
  File "/home/churchyard/tmp/cornice.ext.sphinx/cornice_sphinx/__init__.py", line 78, in run
    app.main({})
  File "/home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/__init__.py", line 11, in main
    return config.make_wsgi_app()
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py27/lib/python2.7/site-packages/pyramid/config/__init__.py", line 891, in make_wsgi_app
    self.commit()
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py27/lib/python2.7/site-packages/pyramid/config/actions.py", line 152, in commit
    self.action_state.execute_actions(introspector=self.introspector)
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py27/lib/python2.7/site-packages/pyramid/config/actions.py", line 293, in execute_actions
    action = next(action_iter, None)
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py27/lib/python2.7/site-packages/pyramid/config/actions.py", line 484, in resolveConflicts
    raise ConfigurationConflictError(conflicts)
ConfigurationConflictError: Conflicting configuration actions
  For: ('view', None, '', 'collection_thing_service', 'a9204cf84fd843dac4e2dad4c4b72fed')
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)
  For: ('view', None, '', 'collection_thing_service', 'b1b988966518e8d3f972d463b864549f')
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)

----------------------------------------------------------------------
Ran 4 tests in 0.203s

FAILED (errors=1)
ERROR: InvocationError for command '/home/churchyard/tmp/cornice.ext.sphinx/.tox/py27/bin/nosetests' (exited with code 1)
py34 create: /home/churchyard/tmp/cornice.ext.sphinx/.tox/py34
py34 installdeps: nose, mock, Sphinx
py34 inst: /home/churchyard/tmp/cornice.ext.sphinx/.tox/dist/cornice_sphinx-0.4.dev0.zip
py34 installed: alabaster==0.7.12,Babel==2.6.0,certifi==2018.11.29,chardet==3.0.4,cornice==3.4.3,cornice-sphinx==0.4.dev0,docutils==0.14,hupper==1.4.2,idna==2.7,imagesize==1.1.0,Jinja2==2.10,MarkupSafe==1.1.0,mock==2.0.0,nose==1.3.7,packaging==18.0,PasteDeploy==2.0.0,pbr==5.1.1,plaster==1.0,plaster-pastedeploy==0.6,Pygments==2.3.0,pyparsing==2.3.0,pyramid==1.10.1,pytz==2018.7,requests==2.20.1,simplejson==3.16.0,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.8.2,sphinxcontrib-websupport==1.1.0,translationstring==1.3,typing==3.6.6,urllib3==1.24.1,venusian==1.1.0,WebOb==1.8.4,zope.deprecation==4.4.0,zope.interface==4.6.0
py34 run-test-pre: PYTHONHASHSEED='3467803108'
py34 runtests: commands[0] | python --version
Python 3.4.9
py34 runtests: commands[1] | nosetests
..E.
======================================================================
ERROR: test_string_validator_resolved (tests.TestServiceDirective)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/churchyard/tmp/cornice.ext.sphinx/tests/__init__.py", line 48, in test_string_validator_resolved
    ret = self.directive.run()
  File "/home/churchyard/tmp/cornice.ext.sphinx/cornice_sphinx/__init__.py", line 78, in run
    app.main({})
  File "/home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/__init__.py", line 11, in main
    return config.make_wsgi_app()
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py34/lib/python3.4/site-packages/pyramid/config/__init__.py", line 891, in make_wsgi_app
    self.commit()
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py34/lib/python3.4/site-packages/pyramid/config/actions.py", line 152, in commit
    self.action_state.execute_actions(introspector=self.introspector)
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py34/lib/python3.4/site-packages/pyramid/config/actions.py", line 293, in execute_actions
    action = next(action_iter, None)
  File "/home/churchyard/tmp/cornice.ext.sphinx/.tox/py34/lib/python3.4/site-packages/pyramid/config/actions.py", line 484, in resolveConflicts
    raise ConfigurationConflictError(conflicts)
pyramid.exceptions.ConfigurationConflictError: Conflicting configuration actions
  For: ('view', None, '', 'collection_thing_service', 'a9204cf84fd843dac4e2dad4c4b72fed')
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)
  For: ('view', None, '', 'collection_thing_service', 'b1b988966518e8d3f972d463b864549f')
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)
    Line 42 of file /home/churchyard/tmp/cornice.ext.sphinx/tests/dummy/views.py:
        config.add_cornice_resource(thing_resource)

----------------------------------------------------------------------
Ran 4 tests in 0.308s

FAILED (errors=1)
ERROR: InvocationError for command '/home/churchyard/tmp/cornice.ext.sphinx/.tox/py34/bin/nosetests' (exited with code 1)
flake8 create: /home/churchyard/tmp/cornice.ext.sphinx/.tox/flake8
flake8 installdeps: flake8
flake8 inst: /home/churchyard/tmp/cornice.ext.sphinx/.tox/dist/cornice_sphinx-0.4.dev0.zip
flake8 installed: alabaster==0.7.12,Babel==2.6.0,certifi==2018.11.29,chardet==3.0.4,cornice==3.4.3,cornice-sphinx==0.4.dev0,docutils==0.14,flake8==3.6.0,hupper==1.4.2,idna==2.7,imagesize==1.1.0,Jinja2==2.10,MarkupSafe==1.1.0,mccabe==0.6.1,packaging==18.0,PasteDeploy==2.0.0,plaster==1.0,plaster-pastedeploy==0.6,pycodestyle==2.4.0,pyflakes==2.0.0,Pygments==2.3.0,pyparsing==2.3.0,pyramid==1.10.1,pytz==2018.7,requests==2.20.1,simplejson==3.16.0,six==1.11.0,snowballstemmer==1.2.1,Sphinx==1.8.2,sphinxcontrib-websupport==1.1.0,translationstring==1.3,urllib3==1.24.1,venusian==1.1.0,WebOb==1.8.4,zope.deprecation==4.4.0,zope.interface==4.6.0
flake8 run-test-pre: PYTHONHASHSEED='3467803108'
flake8 runtests: commands[0] | flake8 cornice_sphinx
cornice_sphinx/__init__.py:270:40: F821 undefined name 'unicode'
ERROR: InvocationError for command '/home/churchyard/tmp/cornice.ext.sphinx/.tox/flake8/bin/flake8 cornice_sphinx' (exited with code 1)
____________________________________________________________________________________________________________ summary ____________________________________________________________________________________________________________
ERROR:   py27: commands failed
ERROR:   py34: commands failed
ERROR:   flake8: commands failed
```

--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -4,6 +4,8 @@
 import mock
 from unittest import TestCase
 
+from pyramid import testing as pyramid_testing
+
 from cornice_sphinx import rst2html, ServiceDirective
 
 
@@ -20,6 +22,16 @@ class TestServiceDirective(TestCase):
 
     def setUp(self):
         super(TestServiceDirective, self).setUp()
+        config = pyramid_testing.setUp()
+
+        def Configurator(settings):
+            return config
+
+        # We need to mock the Configurator, otherwise each test uses the global registry to create
+        # test resources. This leads to an error with Cornice >= 1.4, where resources conflicts
+        # became an error.
+        self.config_patch = mock.patch('tests.dummy.Configurator', Configurator)
+        self.config_patch.start()
         param = mock.Mock()
         param.document.settings.env.new_serialno.return_value = 1
 
@@ -27,6 +39,11 @@ class TestServiceDirective(TestCase):
             'test', [], {}, [], 1, 1, 'test', param, 1)
         self.directive.options['app'] = 'tests.dummy'
         self.directive.options['services'] = ['users', "thing_service"]
+
+    def tearDown(self):
+        self.config_patch.stop()
+        pyramid_testing.tearDown()
+        super(TestServiceDirective, self).tearDown()
 
     def test_module_reload(self):
         self.directive.options['app'] = None

